### DISCUSSION TITLE
How to run the simulation without rendering?

### DISCUSSION THREAD

**Discussion by SHITIANYU-hue** [praveen-palanisamy/macad-gym#46] [Q&A]
I tried to run the basic_agent.py without rendering. I set all the render=False in the environment [https://github.com/praveen-palanisamy/macad-gym/blob/master/src/macad_gym/envs/homo/ncom/inde/po/intrx/ma/stop_sign_3c_town03.py](url). But it will occur this error.

![image](https://user-images.githubusercontent.com/38762389/92312358-d3bc9480-ef8d-11ea-9073-2625edff3ba7.png)

**Comment by praveen-palanisamy** [comment]
Hi @SHITIANYU-hue , 
Which platform are you running on and what's the GPU driver version? 
Looks like you are on an embedded computer with reduced OpenGL capabilities? Or is it just mesa & libEGL limitation?

**Comment by praveen-palanisamy** [comment]
For offscreen rendering, please make sure you are using the (Nvidia) GPU driver accelerated libGL. You may not be able to render (even offscreen) with software-only GL (Mesa). You need a capable hardware (GPU) accelerated libGL capabilities.

You can try using virtualGL & a VNC setup. You may find [this gist](https://gist.github.com/cyberang3l/422a77a47bdc15a0824d5cca47e64ba2#file-how-to-setup-virtualgl-and-turbovnc-on-ubuntu-md) useful for setting up virtualGL.

**Comment by SHITIANYU-hue** [comment]
> Hi @SHITIANYU-hue ,
> Which platform are you running on and what's the GPU driver version?
> Looks like you are on an embedded computer with reduced OpenGL capabilities? Or is it just mesa & libEGL limitation?

Thanks for your reply! This is my GPU version:
![image](https://user-images.githubusercontent.com/38762389/92332782-48ef9e80-f04e-11ea-8c5a-275948e3d0cc.png)

Actually, I run the experiment on the Alienware aurora computer.
